ui/src/shared/components/TimeRangeDropdown.js

```diff
@@ -52,33 +44,33 @@ class TimeRangeDropdown extends Component {
     return selected ? selected.inputValue : 'Custom'
   }
 
-  handleClickOutside() {
+  handleClickOutside = () => {
     this.setState({isOpen: false})
   }
 
-  handleSelection(timeRange) {
+  handleSelection = timeRange => () => {
     this.props.onChooseTimeRange(timeRange)
     this.setState({isOpen: false})
   }
 
-  toggleMenu() {
+  toggleMenu = () => {
     this.setState({isOpen: !this.state.isOpen})
   }
 
-  showCustomTimeRange() {
+  showCustomTimeRange = () => {
     this.setState({isCustomTimeRangeOpen: true})
   }
 
-  handleApplyCustomTimeRange(customTimeRange) {
-    this.setState({customTimeRange})
-    this.handleSelection({...customTimeRange})
+  handleApplyCustomTimeRange = customTimeRange => {
+    this.props.onChooseTimeRange({...customTimeRange})
+    this.setState({customTimeRange, isOpen: false})
```

Comment:
Interesting. The logic looks different, but it works the same (if not better) in the UI. Do you think this was broken to begin with? There's a bit of a bug in this area of the code. See: #1871